presto-session-property-managers/src/test/java/io/prestosql/plugin/session/db/TestingDbSpecsProvider.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.session.db;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.plugin.session.SessionMatchSpec;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+/**
+ * Test implementation of DbSpecsProvider.
+ * It is initialized with a loadingFunction, which is invoked every time
+ * get() method is called.
+ */
+public class TestingDbSpecsProvider
+        implements DbSpecsProvider
+{
+    private Supplier<List<SessionMatchSpec>> loadingFunction;
+    private AtomicReference<List<SessionMatchSpec>> sessionMatchSpecs = new AtomicReference<>(ImmutableList.of());
+    private AtomicBoolean isDestroyed = new AtomicBoolean(false);
+
+    @Override
+    public void initialize(Supplier<List<SessionMatchSpec>> loadingFunction)
+    {
+        this.loadingFunction = loadingFunction;
+    }
+
+    @Override
+    public void destroy()
+    {
+        isDestroyed.compareAndSet(false, true);
+    }
+
+    @Override
+    public List<SessionMatchSpec> get()
+    {
+        if (isDestroyed.get()) {
+            throw new RuntimeException("Provider already destroyed");
+        }
+        try {
+            List<SessionMatchSpec> updatedSpecs = loadingFunction.get();
+            this.sessionMatchSpecs.set(updatedSpecs);
+        }
+        catch (Exception e) {
+            // Swallow exceptions
```

Comment:
Why?

Reply:
In the actual implementation (RefreshingDbSpecsProvider), we log the error and update a failure counter. We can do something similar here too, if required. I've added a `log.error()` statement in the updated PR.